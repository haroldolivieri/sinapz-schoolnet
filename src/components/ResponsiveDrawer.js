import Drawer from '@material-ui/core/Drawer';
import SideDrawer from './SideDrawer';
import React from 'react';

const ResponsiveDrawer = ({navDrawerOpened, toggleDrawer}) => (
    <Drawer open={navDrawerOpened} onClose={toggleDrawer(false)}>
        <div
            onClick={toggleDrawer(false)}
            onKeyDown={toggleDrawer(false)}
        >
            <div className="only-mobyle">
                <SideDrawer />
            </div>
        </div>
    </Drawer>
)

export default ResponsiveDrawer