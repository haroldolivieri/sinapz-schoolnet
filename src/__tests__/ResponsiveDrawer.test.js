import React from 'react'
import ResponsiveDrawer from '../components/ResponsiveDrawer'
import { shallow } from 'enzyme'
import { Drawer } from '@material-ui/core'

describe('Responsive Drawer', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ResponsiveDrawer />)
    expect(wrapper).toMatchSnapshot()
  })

  it('call close method when drawer is closed', () => {
    const onCloseMock = jest.fn()
    const wrapper = shallow(
      <ResponsiveDrawer opened={false} onClose={onCloseMock} />
    )

    wrapper.find(Drawer).simulate('close')

    expect(onCloseMock).toHaveBeenCalledTimes(1)
  })
})
