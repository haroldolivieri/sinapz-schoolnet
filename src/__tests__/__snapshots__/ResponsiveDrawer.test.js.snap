// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Responsive Drawer renders correctly 1`] = `
<WithStyles(Drawer)
  id="drawer"
  open={false}
>
  <div
    className="only-mobyle"
  >
    <withRouter(Connect(SideDrawer)) />
  </div>
</WithStyles(Drawer)>
`;
